2025-03-25 22:13:36,759 : INFO : Start of Prediction
2025-03-25 22:13:36,759 : INFO : run_id:2025-03-25_221336_969656024
2025-03-25 22:13:36,759 : INFO : Start of Data Load, validation and transformation
2025-03-25 22:13:36,759 : INFO : Start of Archiving Old Rejected Files...
2025-03-25 22:13:36,759 : INFO : End of Archiving Old Rejected Files...
2025-03-25 22:13:36,759 : INFO : Start of Archiving Old Validated Files...
2025-03-25 22:13:36,762 : INFO : End of Archiving Old Validated Files...
2025-03-25 22:13:36,762 : INFO : Start of Archiving Old Processed Files...
2025-03-25 22:13:36,763 : INFO : End of Archiving Old Processed Files...
2025-03-25 22:13:36,764 : INFO : Start of Archiving Old Result Files...
2025-03-25 22:13:36,766 : INFO : End of Archiving Old Result Files...
2025-03-25 22:13:36,766 : INFO : Start of Reading values From Schema...
2025-03-25 22:13:36,766 : INFO : End of Reading values From Schema...
2025-03-25 22:13:36,766 : INFO : Start of Validating Column Length...
2025-03-25 22:13:36,767 : INFO : End of Validating Column Length...
2025-03-25 22:13:36,767 : INFO : Start of Validating Missing Values...
2025-03-25 22:13:36,767 : INFO : End of Validating Missing Values...
2025-03-25 22:13:36,767 : INFO : Start of Replacing Missing Values with NULL...
2025-03-25 22:13:36,767 : INFO : End of Replacing Missing Values with NULL...
2025-03-25 22:13:36,767 : INFO : Start of Creating Table...
2025-03-25 22:13:36,768 : INFO : Opened prediction database successfully
2025-03-25 22:13:36,780 : INFO : CREATE TABLE prediction_raw_data_t column_name
2025-03-25 22:13:36,785 : INFO : ALTER TABLE prediction_raw_data_t ADD COLUMN
2025-03-25 22:13:36,790 : INFO : ALTER TABLE prediction_raw_data_t ADD COLUMN
2025-03-25 22:13:36,796 : INFO : ALTER TABLE prediction_raw_data_t ADD COLUMN
2025-03-25 22:13:36,802 : INFO : ALTER TABLE prediction_raw_data_t ADD COLUMN
2025-03-25 22:13:36,808 : INFO : ALTER TABLE prediction_raw_data_t ADD COLUMN
2025-03-25 22:13:36,813 : INFO : ALTER TABLE prediction_raw_data_t ADD COLUMN
2025-03-25 22:13:36,819 : INFO : ALTER TABLE prediction_raw_data_t ADD COLUMN
2025-03-25 22:13:36,824 : INFO : ALTER TABLE prediction_raw_data_t ADD COLUMN
2025-03-25 22:13:36,825 : INFO : End of Creating Table...
2025-03-25 22:13:36,826 : INFO : Opened prediction database successfully
2025-03-25 22:13:36,827 : INFO : Start of Inserting Data into Table...
2025-03-25 22:13:36,827 : INFO : End of Inserting Data into Table...
2025-03-25 22:13:36,827 : INFO : Start of Exporting Data into CSV...
2025-03-25 22:13:36,827 : INFO : Opened prediction database successfully
2025-03-25 22:13:36,828 : INFO : End of Exporting Data into CSV...
2025-03-25 22:13:36,828 : INFO : Start of Moving Processed Files...
2025-03-25 22:13:36,829 : INFO : End of Moving Processed Files...
2025-03-25 22:13:36,830 : INFO : End of Data Load, validation and transformation
2025-03-25 22:13:36,831 : INFO : Start of Preprocessing...
2025-03-25 22:13:36,831 : INFO : Start of reading dataset...
2025-03-25 22:13:36,835 : INFO : End of reading dataset...
2025-03-25 22:13:36,835 : INFO : Start of feature encoding...
2025-03-25 22:13:36,857 : INFO : End of feature encoding...
2025-03-25 22:13:36,857 : INFO : Start of Droping Columns...
2025-03-25 22:13:36,858 : INFO : End of Droping Columns...
2025-03-25 22:13:36,858 : INFO : Start of finding missing values...
2025-03-25 22:13:36,858 : INFO : End of finding missing values...
2025-03-25 22:13:36,858 : INFO : Start of building final predictset...
2025-03-25 22:13:36,862 : INFO : End of building final predictset...
2025-03-25 22:13:36,863 : INFO : End of Preprocessing...
2025-03-25 22:13:36,863 : INFO : Start of Load Model
2025-03-25 22:13:36,863 : INFO : Model File KMeans loaded
2025-03-25 22:13:36,863 : INFO : End of Load Model
2025-03-25 22:13:36,865 : ERROR : Unsuccessful End of Prediction
Traceback (most recent call last):
  File "D:\Major Project\Employee_Retention_ML-main\apps\prediction\predict_model.py", line 27, in batch_predict_from_model
    clusters = kmeans.predict(self.X.drop(['empid'],axis=1))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\KIIT\AppData\Local\Programs\Python\Python312\Lib\site-packages\sklearn\cluster\_kmeans.py", line 1085, in predict
    X = self._check_test_data(X)
        ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\KIIT\AppData\Local\Programs\Python\Python312\Lib\site-packages\sklearn\cluster\_kmeans.py", line 944, in _check_test_data
    X = validate_data(
        ^^^^^^^^^^^^^^
  File "C:\Users\KIIT\AppData\Local\Programs\Python\Python312\Lib\site-packages\sklearn\utils\validation.py", line 2944, in validate_data
    out = check_array(X, input_name="X", **check_params)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\KIIT\AppData\Local\Programs\Python\Python312\Lib\site-packages\sklearn\utils\validation.py", line 1130, in check_array
    raise ValueError(
ValueError: Found array with 0 sample(s) (shape=(0, 9)) while a minimum of 1 is required by KMeans.
